lib/session/call-session.js

```diff
@@ -796,7 +796,9 @@ class CallSession extends Emitter {
         }
       }
 
-      if (0 === this.tasks.length && this.requestor instanceof WsRequestor && !this.callGone) {
+      if (0 === this.tasks.length && ((this.requestor instanceof WsRequestor && !this.callGone) ||
+        // Tasks are all done, but dialog is not establish yet, waiting for cancel/handup the call.
+        !this.dlg)) {
```

Comment:
Are you sure that this won't hit cases other than the outdial failure?  What about an incoming call that an app simply uses a sip:decline verb on?  In that case when the app is over there has never been a dialog.  Can you make sure that case is handled appropriately?

Reply:
you are right, add fix to outbound call only.